app/main.py

```diff
@@ -1,34 +1,86 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self,
+                 start: tuple,
+                 end: tuple,
+                 is_drowned: bool = False) -> None:
+        self.is_drowned = is_drowned
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+        if start == end:
+            self.decks = [Deck(*start), ]
+        if start[0] == end[0]:
+            self.decks = [Deck(start[0], i)
+                          for i in range(start[1], end[1] + 1)]
+        if start[1] == end[1]:
+            self.decks = [Deck(i, start[1])
+                          for i in range(start[0], end[0] + 1)]
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def get_deck(self, row: int, column: int) -> Deck:
```

Comment:
It can return None also

Reply:
I think it can't return None i call  **get_deck** only after cheking
        for coords, ship in self.field.items():
            if location in coords:
                return ship.fire(ship.get_deck(*location))

Reply:
Yes, but this function can return None. We should not think about where this function is used now; we must predict how this function can be used in different cases.